powerstore/datasource_snapshotrule.go

```diff
@@ -0,0 +1,282 @@
+							Computed:            true,
+						},
+						"nas_access_type": schema.StringAttribute{
+							Description:         "The NAS filesystem snapshot access method for snapshot rule.",
+							MarkdownDescription: "The NAS filesystem snapshot access method for snapshot rule.",
+							Computed:            true,
+						},
+						"is_read_only": schema.BoolAttribute{
+							Description:         "Indicates whether this snapshot rule can be modified.",
+							MarkdownDescription: "Indicates whether this snapshot rule can be modified.",
+							Computed:            true,
+						},
+						"managed_by": schema.StringAttribute{
+							Description:         "The entity that owns and manages the instance.",
+							MarkdownDescription: "The entity that owns and manages the instance.",
+							Computed:            true,
+						},
+						"managed_by_id": schema.StringAttribute{
+							Description:         "The unique id of the managing entity.",
+							MarkdownDescription: "The unique id of the managing entity.",
+							Computed:            true,
+						},
+						"interval_l10n": schema.StringAttribute{
+							Description:         "Localized message string corresponding to interval",
+							MarkdownDescription: "Localized message string corresponding to interval",
+							Computed:            true,
+						},
+						"timezone_l10n": schema.StringAttribute{
+							Description:         "Localized message string corresponding to timezone.",
+							MarkdownDescription: "Localized message string corresponding to timezone.",
+							Computed:            true,
+						},
+						"days_of_week_l10n": schema.ListAttribute{
+							ElementType:         types.StringType,
+							Description:         "Localized message array corresponding to days_of_week",
+							MarkdownDescription: "Localized message array corresponding to days_of_week",
+							Computed:            true,
+						},
+						"nas_access_type_l10n": schema.StringAttribute{
+							Description:         "Localized message string corresponding to nas_access_type.",
+							MarkdownDescription: "Localized message string corresponding to nas_access_type.",
+							Computed:            true,
+						},
+						"managed_by_l10n": schema.StringAttribute{
+							Description:         "Localized message string corresponding to managed_by.",
+							MarkdownDescription: "Localized message string corresponding to managed_by.",
+							Computed:            true,
+						},
+						"policies": schema.ListNestedAttribute{
+							Description:         "List of the protection policies that are associated with the snapshot_rule.",
+							MarkdownDescription: "List of the protection policies that are associated with the snapshot_rule..",
+							Computed:            true,
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"id": schema.StringAttribute{
+										Description:         "Unique identifier of the policy.",
```

Comment:
I think its better to specify protection policy explicitly than policy. Applicable for all instances.